### ISSUE TITLE
Error when using BSON Binary type as _id field

### ISSUE THREAD

**Issue by vbuzzano** [balderdashy/sails-mongo#228]
I noticed that when i try to insert a new document with a ._id of type BSON Binary, mongodb failed with an error:

TypeError: Object #<Object> has no method 'value'
  at packElement (/home/ubuntu/workspace/node_modules/sails-mongo/node_modules/mongodb/node_modules/bson/lib/bson/bson.js:773:26)

I debug to know why and found that in the file document.js (sails-mongo) in the prototype method normalizedId, you swap .id to ._id by deep cloning:

    values._id = _.cloneDeep(values.id);

Then BSON object does not keep value and position funciton wich is need by mongodb bson/binary.js

Why do you use cloneDeep, Would it be better to just assign variable ?
    values._id = values.id;

**Comment by vkopitsa** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @vbuzzano.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
